src/pages/categories/CategoriesHeader.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react'
+import {
+  Box,
+  Typography,
+  Button,
+  TextField,
+  InputAdornment
+} from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import { useNavigate } from 'react-router-dom'
+
+interface CategoriesHeaderProps {
+  onSearch: (searchQuery: string) => void
+}
+
+const CategoriesHeader: React.FC<CategoriesHeaderProps> = ({ onSearch }) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const navigate = useNavigate()
+
+  const handleShowOffers = () => {
+    navigate('/categories/subjects/find-offers')
```

Comment:
```suggestion
    navigate('/categories/${id}/subjects/find-offers')
```
And change routing for this